### ISSUE TITLE
Using custom Makefile produces "file or directory not found" error

### ISSUE THREAD

**Issue by JoeKennedy** [mfine/heroku-buildpack-stack#20]
I'm using a custom `Makefile` in order to build with `happy` and `alex` for a Yesod project, and getting the following error when pushing my yesod project for the first time:

```
> git push heroku master
Counting objects: 98, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (96/96), done.
Writing objects: 100% (98/98), 38.30 KiB | 0 bytes/s, done.
Total 98 (delta 46), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Haskell app detected
remote: -----> 2017-05-12_04:39:02 Starting...
remote: -----> 2017-05-12_04:39:02 Exporting config vars
remote: -----> 2017-05-12_04:39:02 Installing libgmp-6.0.0a into cache
remote: ######################################################################## 100.0%
remote: -----> 2017-05-12_04:39:02 Restoring 6.0.0a files from cache
remote: -----> 2017-05-12_04:39:02 Downloading stack-1.4.0
remote: ######################################################################## 100.0%
remote: -----> 2017-05-12_04:39:03 Restoring stack-1.4.0
remote: -----> 2017-05-12_04:39:03 Substituting stack vars
remote: -----> 2017-05-12_04:39:03 Running stack
remote: -----> 2017-05-12_04:39:03 Makefile detected
remote: make: Entering directory `/tmp/build_01b7b5044fb43461563e4b58849a910e'

...

remote: make: Leaving directory `/tmp/build_01b7b5044fb43461563e4b58849a910e'
remote: -----> 2017-05-12_05:03:30 Making stack binaries available
remote: -----> 2017-05-12_05:03:30 Caching .stack-work
remote: rsync: change_dir "/app/.stack-work" failed: No such file or directory (2)
remote: rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1183) [sender=3.1.0]
remote:  !     Push rejected, failed to compile Haskell app.
remote:
remote:  !     Push failed
remote: Verifying deploy....
remote:
remote: !	Push rejected to fantasygameofthrones.
remote:
To https://git.heroku.com/fantasygameofthrones.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/fantasygameofthrones.git'
```

I believe I've managed to fix this in [my fork](https://github.com/JoeKennedy/heroku-buildpack-stack), and I'll start a pull request for it. It seems like the `make` command just needs to be passed a different option.

**Comment by mfine** [comment]
Hey @JoeKennedy, thanks for the issue! I think there's an issue in how the Makefile was getting invoked - it was going to the wrong directory. I've got a simple fix for it in https://github.com/mfine/heroku-buildpack-stack/pull/22. Would you mind trying that?  Url would be ```https://github.com/mfine/heroku-buildpack-stack.git#fix-make```. Alternatively, if you have a repo I could clone and try, that would be awesome. Thanks for your help!

**Comment by mfine** [comment]
Should be fixed by #22. Please reopen if that doesn't work out.
